### ISSUE TITLE
is this project dead?

### ISSUE THREAD

**Issue by cool-firer** [rethinkdb/rethinkdb#6747]
is RethinkDB dead?

**Comment by itamarjp** [comment]
not yet

**Comment by jsieler** [comment]
but it looks like it is...

**Comment by v3ss0n** [comment]
in coma

**Comment by Extarys** [comment]
It seems that contributors don't have time to help the project. There is a couple of things being worked on but it's slow. There no huge team behind the project and people have other stuff going on. 

It's sad because RethinkDB have very good potential, but until a miracle person with a ton of cash can pay people to work on it, development will be slow.

**Comment by grantcarthew** [comment]
This project is amazing however the lack of interest is having a knock-on effect.

I was working with RethinkDB heavily in the past. I wrote [rethinkdb-job-queue](https://github.com/grantcarthew/node-rethinkdb-job-queue) because I had a need for a job or task queue and wanted to build it on RethinkDB.

I have since moved onto MongoDB because this project is not popular and doesn't get the love it deserves. This is a common theme I think you will find.

Not only is the database not getting love, its community is deserting it. Myself being a part of that.

I doubt it will awake from its coma as @v3ss0n has called it. I will not be moving back to RethinkDB even if it does get love. I have moved on. It's sad.

For reference: https://db-engines.com/en/ranking_trend/document+store

**Comment by 4ley** [comment]
That's sad. But seems like someone is working on it. There is a beta release from 1st of April. So petty that there is no more attention. The concept is great.

**Comment by jsieler** [comment]
The probably nearest to RethinkDB is ArangoDB. What i liked on RethinkDB is its simplicity, and even portable run :)

**Comment by cool-firer** [comment]
what a pity, I like RethinkDB

**Comment by piusalfred** [comment]
I like RethinkDB, It got a big potential

**Comment by chovy** [comment]
They guys who forked it should have never rejoined, once they did that it was doomed to fail.

**Comment by traverseda** [comment]
Yeah, not being able to install it on any of the OS's I use is making it pretty hard to even play with the project.

**Comment by TheNetJedi** [comment]
@traverseda 

I was able to get it to work using Docker

**Comment by srh** [comment]
I have been insanely busy and neglectful, and too stressed out to even think.  But I have forced my schedule open.  I still have one bit of work to do first.  But then I will be finishing my implementation of RethinkDB with a RocksDB backend.  And I plan to get that released.

Then my plan is then to make a non-free FoundationDB-based clustering backend to RethinkDB and license it to any comers.  Unless anybody wanted to pay me up front to do it open source... but I've been pretty flakey lately, so I can't blame you if you don't want to do that.

There's also some work by Christina Keelen and Linux Foundation people to help get some of the donations to the Linux Foundation put into motion to get the project's immediate needs met.  I'm avoiding that (because making a closed-source FDB-based backend and then getting funding to do other tasks is kind of... awkward) but there are some other people that are doing such stuff.

So I think there is a future for you if you are using RethinkDB.

**Comment by traverseda** [comment]
Do you anticipate the existing clustering solutions continuing to work, as well as the foundationDB? Or is this going to be a more complete rewrite?

**Comment by KittyBot** [comment]
Thanks @srh 

Hey everyone, I'm sorry for hopping in so late, that wont be the case after this. I'm the original community manager from Rethink and have been working with folks throughout the past few years to try and get old loose ends tied up so we can progress with future releases. Unfortunately, as is the case with most open source projects, a lot of us have been too busy to make things move as quickly as possible. For everyone who continues to stick around and make contributions (in any capacity), **thank you** very genuinely. 

The great news is that we finally have progress on the website, slack, and stripe account, which were obstacles that were holding up a lot of the releases. I am also working on a blog post noting the Rethink's current status, to thank our community contributors, partners, etc. so please keep your eyes open for a "What's next" issue as you may be able to help me out there. :)

I've reached out to @gabor-boros who has been helping me and can provide more detailed updates on his end. 

I'm pretty sure I know a lot of you already (if you have any swag, I sent it to you :)), but hello to everyone new, too. 👋

**Comment by jsieler** [comment]
@everyone, i've created Discord channel for RethinkDB. If you feel like you want to help, need help or want to share your experience, feel free to join: https://discord.gg/dhTESz4 (for someone who not using slack, open for suggestions for additional channels).


@srh we all know how it is to make some side projects 👯‍♂ Less on time, family, etc... But, you could at least give people some "feedback" on how things are going before people start discussing on "if the project is dead". Probably that is the problem why people are getting away from RethinkDB (will they ever come back?).

> RethinkDB with a RocksDB backend
So, that means that 2.4 will be based on RocksDB? What is the advantage then if people might use RocksDB directly?

> Then my plan is then to make a non-free FoundationDB-based clustering backend to RethinkDB
So, as i understand, RethinkDB will be able to use multiple backend? Why not to concentrate on just one? Which advantages?

>  and license it to any comers.
Why not just to add Corporate support for whole RethinkDB like NGINX does? They have both versions and business runs well.

> Unless anybody wanted to pay me up front to do it open source...
I wish that could happen, but i wouldn't pay for dead project unless you resurrect it (necromancer) (imho).

> There's also some work by Christina Keelen and Linux Foundation people to help get some of the donations to the Linux Foundation put into motion to get the project's immediate needs met. I'm avoiding that (because making a closed-source FDB-based backend and then getting funding to do other tasks is kind of... awkward) but there are some other people that are doing such stuff.
See example of NGINX. Open source with paid support and additional modules. You also could offer SaaS service as well like others do,


@KittyBot thanks for your reply. I think a lot of people were looking for an answer what is going on here. I just hope that your message will be not last on here and we will really see some progress on RethinkDB. 

Good news to know that you have some progress on website and stripe account. I think if you're going set-up some "donation" account, there will be plenty lot people (including me) who will donate few bucks per month, so you might even arrange some coders (or pay @srh) onto your side or somehow to speed-up development of version 2.4.

I wish you best luck and hope to see new version soon! 👍

**Comment by gabor-boros** [comment]
Hello everyone,

Thank you @KittyBot for introducing me.

@chovy, I'm one of the guys who forked the project and I was here when we merged it back. Although it seems that the development slowed when we did the merge, but we felt we had no chance to survive without the community’s support. Thanks to your contributions to Rebirth DB, The Linux Foundation generously allowed us to merge the project back so we are one community again.

As @KittyBot mentioned, we were able to build the RethinkDB website again and set up continuous deployments for it, which allows us to publish new blog posts, configure the stripe account, and of course, update the content which is a bit outdated.

Also, I'm one of the maintainers of the Python client. Nowadays, I had busy days and I had not too much time contributing to the project but it will change.

At the moment with @KittyBot we are collecting every blocker and obstacle which need to be eliminated, and we are setting up both short and long term plans to keep the community alive and loop more guys into the development of the DB.

@jsieler, I'm glad that you created the discord channel for RethinkDB, but at this point, I would like to note that we got back the access to Slack. To set up a self-invite flow for Slack is also on our roadmap so we will be able to keep the community at one place. (or at least try it)

The communication was not our strength with the future of the project. We noted this and this will change in the future.

I would like to thank you for those who contributed to the project since it was open sourced and for those who stayed with us.

**Comment by srh** [comment]
> Do you anticipate the existing clustering solutions continuing to work, as well as the foundationDB? Or is this going to be a more complete rewrite?

The FDB-backend stuff won't be in RethinkDB's repo.  So RethinkDB will continue to be the same.

> So, that means that 2.4 will be based on RocksDB?

No, I'd think that would be post-2.4, since it's useful to get out a more stable version for 2.4, then release that as 2.5, if people like that change.

> What is the advantage then if people might use RocksDB directly?

RocksDB is an internal implementation detail, it just means better performance (because RethinkDB's storage engine is craptacular), and avoids some of the bugs we've been running into (like a mysterious gc_entry_t crash, and a memory leak in 2.3.5 and 2.3.6 that some users have been seeing).  It doesn't have RethinkDB's feature set, so it can't be used directly as a replacement for RethinkDB.

> Do you anticipate the existing clustering solutions continuing to work, as well as the foundationDB?

Yes, because the FDB backend stuff is separate from RethinkDB.

> So, as i understand, RethinkDB will be able to use multiple backend? Why not to concentrate on just one? Which advantages?

RethinkDB will be unchanged.  I am concentrating on one.  The advantages... I'll put my marketing copy elsewhere.

> Why not just to add Corporate support for whole RethinkDB like NGINX does? They have both versions and business runs well.

I am making ReQL-on-FDB because I think it is useful to do so, and it would be a useful tool for people running RethinkDB clusters.

**Comment by ChrisTalman** [comment]
Many thanks for the update, @srh. I'm sorry to hear about the stress - personal wellbeing is paramount.

Concerning funding for ReQL-on-FoundationDB, I recall that crowdfunding was considered for it at one point. It might be worth a shot before going down the licensing path. It's preferable, in principle, for the project to be open source and available for all to use at no cost, _if viable_. It could also help to revive RethinkDB as a whole, by being accessible to newcomers and returning users, no questions asked. Crowdfunding could be a little less pressuring than direct funding, given that it is common for crowdfunded projects to take some time to come to fruition, and people know that. It'd require some organisation to get off the ground, including spreading the message among the remaining Rethink community. Perhaps @KittyBot and others could lend a hand with this? If it failed to raise enough funding to support the project, the funds could be refunded, and the project could fall back on the licencing approach. But, as I say, it seems to be worth a shot, for the chance of keeping the flame of open source and open access alive.

**Comment by NetOperatorWibby** [comment]
I just want to chime in here and say that I’m really enjoying RethinkDB for one of my projects. I was looking at FoundationDB but it was too confusing to setup and use. I was looking to migrate away from the frustration of MongoDB’s lack of information as to why a database wasn’t behaving, for example.

Had no idea of the history behind Rethink but I’m glad it’s not dead yet, especially since I just moved to it mere days ago.

Also good to see communication will be improving, I’ll be bugging y’all with questions soon.

**Comment by traverseda** [comment]
Honestly I'll be happy if it can just be compiled under modern boost.

There was a time I was running a cluster across my phone and laptop using zerotier and termux. That was nice.

**Comment by ChrisTalman** [comment]
@NetOperatorWibby
> I was looking at FoundationDB but it was too confusing to setup and use.

I've also looked into Foundation and found its administration a little daunting. There was quite a detailed discussion about ReQL-on-FoundationDB in the [Spectrum community](https://spectrum.chat/rethinkdb). Unfortunately, I don't recall the precise threads. I think the takeaway was that, while Foundation might not be as straightforward to administrate as Rethink, it's ultimately workable, with the benefits in performance, stability, and features making it worthwhile.

**Comment by thelinuxlich** [comment]
Happy to hear these news!

**Comment by AaronRobinson** [comment]
Yeah me too. I don't want to move away from RethinkDB. @KittyBot @gabor-boros Can I suggest that now you can deploy the website that you add a banner to the home page stating some to the effect that the project is not dead and is actively being resurrected? I use the website daily and I had no idea this conversation had been started until @srh answered one of my old posts and pointe me to it. It would encourage use. On the Discord front - yet another channel? With Slack access back I don't see why we wouldn't revert to that. Spectrum is pretty clunky IMO.

**Comment by WoLfulus** [comment]
I also don't like Spectrum (neither Slack tbh.), discord would be nice though.

**Comment by gabor-boros** [comment]
@AaronRobinson Thank you for your suggestion. 

Last week I spent some time to be able to redeploy the site again on an automated basis. Currently, with @KittyBot, we are revisiting the content and the website related GitHub issues.

My goal for the next week is to redeploy the site. This does not mean that everything will be up-to-date by then, but we will be able to deploy changes.

As I mentioned before, communication was not our strength until now. I would like to inform you about the changes around RethinkDB on a weekly basis (at least I would try it).

Related to Slack, I'm actively working on to open up slack again for everyone who would like to join. I think this will be done by the middle of next week.

**Comment by gabor-boros** [comment]
>I also don't like Spectrum (neither Slack tbh.), discord would be nice though.

@WoLfulus I totally understand this. Although I'm not against discord (really), we have a lot of people at Slack, and we need to get their confidence back. If we switch platform now for communication, I believe, we miss the opportunity to reach those members.

**Comment by AaronRobinson** [comment]
@srh why RocksDB versus the other kv stores?

**Comment by v3ss0n** [comment]
@srh  RocksDB or FDB , if anything breaks realtime-ness of rethinkdb , it will trow many of the remaining users.

The main reason for using rethinkdb were realtime things can be done easily with reql + changefeed .

Please consider many of the main rethinkdb users... 
Jumping between KV Stores are showing the lack of direction too..

**Comment by thelinuxlich** [comment]
"many of the main rethinkdb users", like the project is overflowing of contributions...

Linus Torvalds is always right, talk is cheap, show me the code.

**Comment by AaronRobinson** [comment]
I don’t think it’s jumping between KV stores. We need a new engine and that
will be RocksDB. The FDB stuff will be done privately and be licensed and
not linked to the RethinkDB project.

**Comment by ChrisTalman** [comment]
> The FDB stuff will be done privately and be licensed and
not linked to the RethinkDB project.

I'm hoping the crowdfunding avenue is not out of the question yet.

**Comment by v3ss0n** [comment]
@thelinuxlich  show me the list of users who worried about the backend engine and it quirks more than realtime features . 

I can list Spectrum , and 3 of my main products. 1 of them is growing 200k users 2000 active concurrent users.

**Comment by AaronRobinson** [comment]
Hopefully. But was just making point that there isn’t indecision I can see.
If nobody can/wants to maintain the existing engine then the change isn’t
optional. We’re lucky we have someone that can even do it.

**Comment by thelinuxlich** [comment]
BigchainDB dropped RethinkDB a long time ago due to its limitations. 

@AaronRobinson wrote exactly my thoughts. These "successful" RethinkDB use cases didn't give a damn about the database future. If someone is willing to take it somewhere(it doesn't matter where, what matters is IT IS MOVING), then it's awesome.

**Comment by ChrisTalman** [comment]
@v3ss0n

> RocksDB or FDB , if anything breaks realtime-ness of rethinkdb , it will throw many of the remaining users.

From what I recall of @srh's comments about ReQL-on-FoundationDB, he thought that it might be possible to maintain realtime functionality. But, at that time, he said he was still thinking it through, and thought it might not be available in the initial release. I'm not sure what his thinking is on it at this point. It's probably best to simply wait and see what he says.

In any case, it's important to note that this isn't about breaking realtime functionality. It's about having a reliable technology base upon which the project can move forward.

Unfortunately, the project lacks the resources, in funding and developers, to maintain its own storage engine in a reliable manner that improves functionality and performance well into the future. It's a very demanding undertaking. Thus, it's not a choice between the old engine and a new one. Rather, it's between the continued stagnation of the project, on the one hand, and the continuation of a huge part of the project, ReQL, on the other, through a new storage engine.

FoundationDB is reliably maintained by Apple, enabling the project's limited resources to focus upon ReQL and everything around it. This includes new and improved ReQL commands, features, drivers, and tools, while at the same time benefiting from improved storage engine performance. Therefore, ReQL-on-FoundationDB seems to be the sensible way forward.

**Comment by srh** [comment]
@ChrisTalman It's not out of the question.

**Comment by jsieler** [comment]
Which benefit will it bring by using RocksDB over... ArangoDB which is using same? (https://www.arangodb.com/why-arangodb/rocksdb-storage-engine/)

**Comment by traverseda** [comment]
@jsieler I can answer that. Portability. RocksDb is a library that you can compile for all kinds of platforms, where as arrangoDb will only run on some platforms and you'll need to do a lot more work to port it over.

Really though it's just that rocksDb is a library (embeddable) and arrangoDb isn't.

**Comment by jsieler** [comment]
@traverseda ehm, you probably didn't got my question right. Which benefit it will have for users, to wait for some release of RethinkDb on RocksDb and less known functionality where solid product like arangodb is already there, with same backend?

**Comment by traverseda** [comment]
Ahh, yeah. I mis-read that.

**Comment by ChrisTalman** [comment]
@jsieler
>Which benefit it will have for users, to wait for some release of RethinkDb on RocksDb and less known functionality where solid product like arangodb is already there, with same backend?

I'm not familiar with the specific reasoning behind the ReQL-on-RocksDB project, but, if I were to hazard a guess, it might be that it is a quick way to be able to use ReQL with a more robust storage engine. The existing storage engine has bugs and performance issues. ArangoDB may use RocksDB by default, but it has a fundamentally different query language from ReQL. Indeed, this is true for most databases. ReQL is very unique. I am only aware of one other database with a similar query language, which is FaunaDB. For those who like ReQL, but are justifiably reluctant to rely upon the existing issue-prone storage engine for a project, RocksDB could give them the confidence to use ReQL in a project with peace of mind.

Personally, I'm hoping that the community will be able to raise the funds for the ReQL-on-FoundationDB project. The focus of the Rocks project is to provide a robust base. In contrast, the Foundation project could go much further, allowing for the introduction of new ReQL features, like transactions, which have long been sought, but were a challenging undertaking to implement in the old storage engine.

**Comment by robertjpayne** [comment]
I loved RethinkDB, its query language and the team behind it even once it went under for a time being.

However, in saying that. RethinkDB is unequivocally a dead project for the foreseeable future. Development has been pretty non-existent for over a year and in the database world that is bad bad news.

It's a shame because it was a crazy easy to use database that did safe by default distributed storage right but databases are a hard sell since there are so many amazing free options out there.

@srh if I could really make a suggestion if you want to make $$ is to make a ReQL layer that runs on top of PostgreSQL. It doesn't even have to really conform to SQL that much, just use jsonb columns but the idea there is you can run it against AWS RDS or AWS Aurora or any of the other major cloud providers that offer PostgreSQL as a service.

FoundationDB is great, but I fear it'd ultimately not be overly popular due to the know-how of running a FDB cluster required.

If I wished any one thing could live on past this project in another life it'd be the ReQL API against a standard SQL database with json column storage.

**Comment by tinco** [comment]
Why not "just" write a layer that translates between reql and mongodb? As far as I can tell MongoDB has feature parity, so it should be possible to do it without dropping any features. It has a much larger userbase and community than Foundation DB and is commercially maintained. We could probably even get the nice web panel going for it.

Swapping the storage engine to RocksDB sounds more technically involved to me, and would only solve a subset of the issues. Basically it would assume everything else in RethinkDB is working fine.

**Comment by thelinuxlich** [comment]
MongoDB isn't even truly opensource anymore...it's more or less "open core with some gotchas"

**Comment by tinco** [comment]
Ah, that's unfortunate, I didn't know that.

**Comment by jsieler** [comment]
MongoDB isn't even near to what RethinkDB ist. The probably most nearest product would be ArangoDB, but it hat lot more features than RethinkDB and MongoDB together, and also heavier than RethinkDB.

**Comment by robertjpayne** [comment]
@tinco MongoDB has the same issue as FoundationDB which is there is a very limited number of options for hosting.

PostgreSQL is provided by all the major cloud providers in various configuration models where as MongoDB deployments tend to be 3 node clusters which start at a high price.

Unfortunately it'd still be a lot of work to keep performance up with ReQL implemented on top w/o bolting it directly into the core which would require you to host it yourself

**Comment by tinco** [comment]
@jsieler what do you mean by that? What sets Arango and RethinkDB apart from MongoDB? Perhaps there's a usecase I'm missing, but if I look at my RethinkDB apps I see a very clear migration path to MongoDB. Perhaps I'm biased because it's the only other NoSQL DB I've used in production.

@robertjpayne I didn't think of hosting solutions, but I guess today hosting your own DB is quite old fashioned, that's a good point. Even if we could use pgsql as a storage backend, I bet that wouldn't be possible on existing hosting solutions, you'd need a customized pgsql/rethinkdb combination.

**Comment by robertjpayne** [comment]
@tinco it's one thing to host your own query layer "frontend" vs hosting the "backend" storage layer. The front end doesn't have any concerns about distributed computation, transactional issues etc… it simply talks to the storage backend which does all the really hard stuff.

Really a ReQL layer in front of Postgres would be no different than hosting your own node.js app as it would be identical and convert ReQL queries to SQL queries and return data accordingly. The barrier to entry there is waaaay lower than hosting your own database and all the nasties you have to account for there with backup and disaster recovery.

**Comment by tinco** [comment]
Right, but pgsql doesn't have changefeeds, nor does it have easy high availability or horizontal scaling. They are fundamentally different types of database. That is why I'm saying mongodb, because it has all the features rethinkdb has, it would literally be as easy as implementing a translation between rql and mongodb statements. Unless I'm missing something like @jsieler suggests.

**Comment by ChrisTalman** [comment]
@tinco
> it would literally be as easy as implementing a translation between rql and mongodb statements

I don't quite follow how that would be possible. ReQL is similar to a programming language in some regards, even if it ultimately falls short of that qualification, all things considered. It affords a significant degree of flexibility in how queries are constructed. I do not have much experience with MongoDB, but when I have used it, the query language seemed to be restrictive by comparison. Even if this were not the case, and the MongoDB query language was just as flexible, translating from one language to another tends not to be a walk in the park in my limited experience.

**Comment by tinco** [comment]
I have some experience in writing query language serializers, and it is not very difficult, unless there's some sort of fundamental disconnect between how MongoDB and RethinkDB treat queries. I just took a quick look at MongoDB's query language and if anything it seems more powerful. Apparently it's even possible to straight up run Javascript inside MongoDB. And since MongoDB 4 it can do relational queries as well.

A cool thing about declarative programming languages like these query languages is that they usually can be transformed and simplified really easily. Sometimes it might seem like you're constructing a very advanced SQL query, but in the end it's just more of the same thing. You don't introduce these exponential complexities you have in imperative programming languages.

**Comment by NetOperatorWibby** [comment]
So...where can I throw money to make this software viable again?

**Comment by ChrisTalman** [comment]
@NetOperatorWibby 
> So...where can I throw money to make this software viable again?

I don't think there's anywhere in particular to do so at the moment. [The site accepts donations](https://rethinkdb.com/contribute/#donate), but it's not clear where that money will go. It's probably best to simply wait and see whether @srh wishes to try crowdfunding for ReQL-on-FoundationDB, and to support him at that point if he chooses to do so.

**Comment by KittyBot** [comment]
We're reworking the website and donations page.. we will definitely be ready for donations once the tasks I'm working on with @gabor-boros are checked off the list. :)

**Comment by ChrisTalman** [comment]
@tinco Do you know whether it is possible to run an arbitrary query in MongoDB which does not involve any table? For instance, could you add two numbers together? Further, could you do it without the raw JavaScript feature you mentioned? I was not able to determine that queries like this are possible in MongoDB. In my limited use of it, the language seemed to consist of commands that, while perhaps warranting the term 'powerful' in themselves, were rather discrete. It did not seem possible to blend them with one another in the same way that is possible in ReQL.

**Comment by thelinuxlich** [comment]
The conflict functions are impossible to entirely reproduce within Mongo query language.

**Comment by thelinuxlich** [comment]
FYI Apache CouchDB 4 will use FoundationDB under the hood.

**Comment by thelinuxlich** [comment]
There are other databases already using it, like Snowflake.

**Comment by omidkrad** [comment]
RethinkDB is removed from [Azure Marketplace](https://azuremarketplace.microsoft.com/en-us/marketplace/apps?search=RethinkDB) and is marked as DEPRECATED on [Docker Hub](https://hub.docker.com/_/rethinkdb). I think no one will use RethinkDB for a new project today and maybe it is a better idea to merge it with some other open source DB such as FoundationDB.

**Comment by jsieler** [comment]
That's really sad to hear, but... Looking at last "promises" and not clear definition about the feature of "rethinkdb", makes me cry, really. I just see no hope and feature that this datatase will rise after such a fall.. :(

https://bigdata-madesimple.com/18-free-and-widely-used-open-source-nosql-databases/
Looking at RavenDB right now... Looks nice.

**Comment by NetOperatorWibby** [comment]
RavenDB looks interesting but RethinkDB feels intuitive so I’m sticking with it.

**Comment by withinboredom** [comment]
I've been watching this saga play out for years ... every time I start a new project I come back here hoping. Hoping it's still alive, and it is. Albeit, limping along.

I'm excited to see RocksDB being considered, it's a pretty solid storage engine. FWIW, just for giggles, I'm about to use rethinkdb for a new project. Maybe that's a bad idea, maybe not.

**Comment by ChrisTalman** [comment]
I'm still hopeful about the prospects for RethinkDB, because of the opportunities of the ReQL-on-FoundationDB project.

It seems to be the only viable means by which the core of RethinkDB, ReQL, can live on, and not only in terms of stability, but in terms of actually moving the query language forward, and growing the community and ecosystem as a result.

I hope we will eventually have the opportunity to crowdfund that project, so that it can be readily available for the whole community.

But, @srh, who may be able to bring the project to fruition, is one person. If this is to have any chance of transpiring, we must be patient. 👍

**Comment by AaronRobinson** [comment]
Would seem to me to make sense to put all effort into sorting out funding channels.

**Comment by sholladay** [comment]
I'm also interested in funding development to help get things back on track. If anyone knows qualified people who can make some time to work on RethinkDB, could you please put them in touch with me? From the discussions I've had over the past few months, I think we definitely have enough community interest to make it viable. We just need to organize better and make sure people have access to everything, which people in this thread are already working on.

**Comment by ChrisTalman** [comment]
From what I've gathered from following these discussions over the past few years, the only open and active qualified person seems to be @srh, a former employee of RethinkDB. He seems to be working on ReQL-on-RocksDB at the moment. I'm not sure how that's funded. Once that's done, he plans to work on ReQL-on-FoundationDB, hopefully through crowdfunding if the community can raise enough funds. In the past, he's said that he's very busy. [Please see his comment here](https://github.com/rethinkdb/rethinkdb/issues/6747#issuecomment-509846944). Thus, he may not be in a position to be able to respond to this discussion at this time. Besides taking unilateral action on his behalf, which might be unwise, it seems that we have to be patient for the time being. But, once he's ready, I hope the community will be able to fully fund ReQL-on-FoundationDB, which seems to be the best way forward for RethinkDB as a whole.

**Comment by jsieler** [comment]
Why not to start the funding on Kickstarter or like that!? Fontawesome did that well and got a lot of people.

**Comment by sholladay** [comment]
I have a contract with @srh for some new features in the admin panel. But as you said, he is very busy with other commitments. We could use more hands. Money is available for anyone who has the time.

**Comment by KittyBot** [comment]
Just wanted to let you know we're working on a few logistical updates for this week with @gabor-boros and @mglukhovsky. 

We'll also work with LF/CommunityBridge for fundraising and will have updates on that and how you can help, where funds are going, etc. 

https://communitybridge.org/

Thanks again :)

**Comment by KittyBot** [comment]
Setting up the CommunityBridge project is slightly more time consuming than I thought, but I'm chugging along this week. Will continue to keep you posted.

**Comment by AaronRobinson** [comment]
Do you need any help?

**Comment by KittyBot** [comment]
@AaronRobinson Currently everything is in that transitional phase, the stuff that's kept the project from being able to move forward (tracking down credentials to update website, setup fundraising, track status and funds for Stripe, update the project so it's more aligned with other LF projects, including a CoC, CommunityBridge, etc...) so sadly there is not much help I can receive since it's a lot of sorting through backlog and organizing current information. @gabor-boros is helping me on this and we've made a lot of progress. 

We will _absolutely_ need help very soon, though, but we had a case of "too many cooks in the kitchen" over the past few years (I was one of those cooks) so it's sort of reorganizing everything so it can move smoothly go forward. 

TL;DR - No help in this moment, but definitely very soon. Unless @gabor-boros has anything he needs help with. <3 Thank you

**Comment by MichaelFedora** [comment]
Glad to see the project isn't dead -- RethinkDB is my favorite DB to use due to how easy it is to spin up and just *use* it. ReQL is definitely a factor in that, and last time I checked in on this discussion I saw @robertjpayne 's post on how it's dead and therefore focusing on getting ReQL interpreters for other databases would be a better use of time.

I'm not sure where else to post this, but I made a SQLite3 interpreter for (a subset of) ReQL which can be found [here](https://github.com/MichaelFedora/reql-sqlite3). While it's all done in Javascript, which is non-optimal for cross-language use as well as efficiency, it's definitely much more fun to use then creating SQL queries every time you need to access or modify the database. It also seamlessly transitions between the Database query object and doing in-memory operations (such as `map`), with nothing being run until `run()` is called.

It should be pretty easily translatable into other SQL databases (such as Postgre) for use until RethinkDB is back up to date, though it also provides a decent framework for a minimal ReQL implementation (I'm considering building it on top of `levelup` in the future). I will not claim it is stable for production use, however, as I made it in two days.

Everyone should feel free to use/adapt the above, and I'm totally open to helping the RethinkDB team as well, though I'm not sure how JS expertise would fit in that. Regardless I can't wait to see where this database and ReQL goes in the future!

**Comment by gabor-boros** [comment]
Status Update:

With @KittyBot and @srh we are progressing with prerequisites for CommunityBridge and trying to resolve issue #6750 as soon as we can.

**Comment by v3ss0n** [comment]
Very interesting progress! THanks a lot!

**Comment by lsabi** [comment]
Before deciding to which database/architecture to move, wouldn't be good to ask also the community for their needs and if they are willing to use RethinkDB in the future? This would serve for understanding the usage and the needs of users.

Otherwise a public discussion with an appropriate thread and votes of developers and users. Also, this may help identifying problems that may be faced, better splitting/organizing responsibilities and development.

Just suggesting, since in the future I would like to help developing the new database

**Comment by ChrisTalman** [comment]
@lsabi There has been a significant amount of discussion about ReQL-on-FoundationDB [on Spectrum](https://spectrum.chat/rethinkdb). Unfortunately, I can't recall the precise posts and threads where this took place. @srh also wrote a [blog post](http://samuelhughes.com/blog/10.html) touching upon it.

The crux of the matter is that this community does not have enough database developers to reliably maintain and advance its own storage engine. Thus, if the project is to continue, it seems that it must continue without its own storage engine.

@srh, one of the few database developers in the community, has identified FoundationDB, reliably maintained by a major technology company in the form of Apple, as being a viable alternative storage engine on which ReQL can run.

It could be described as a low-level database which was explicitly designed to have layers, like ReQL, built on top of its core. It could afford ReQL users long-term reliability, but also performance benefits, and the potential to add new features to ReQL in the future, such as transactions.

I'm not aware of any viable alternative to FoundationDB. Thus, it seems to be the best way forward for this project.

**Comment by lsabi** [comment]
@ChrisTalman thanks for the explanation. I was just suggesting to understand the needs of users before choosing (but I did not know about the discussion on spectrum, my bad, sorry).

Anyway, it seems a good reason for porting ReQL to FoundationDB.

**Comment by KittyBot** [comment]
We're still blasting out this best practices stuff. Sorry it's so slow! Also getting the financials sorted out with LF and Stripe and will share a blog post with that info asap (let's say ~month to be safe).

**Comment by MrLeap** [comment]
RethinkDB has the best dev ergonomics I've ever seen. All the work every contributor has done is appreciated by this user. Thank you!

**Comment by KittyBot** [comment]
CII Best Practices is complete, CommunityBridge has been submitted and awaiting approval, in touch with Stripe for our first match, and currently working on a blog post... _whew_ we're gettin there... :)

**Comment by aaamitsingh** [comment]
It's very sad to see this project dieing. This is one of my favourite and the best choices for IoT projects.

**Comment by yaneony** [comment]
So, any news?

**Comment by KittyBot** [comment]
Yes, thank you @sholladay you are correct. The goal is to build confidence back up in the community that development will continue and everything will remain on track. 

Good news is that CommunityBridge is all set up (sponsorship button added to the repo up top) and we're wrapping up some other minor things like setting up emails, etc. 

We're also working on a blog post and will be able to share more formally asap!

**Comment by yaneony** [comment]
Sorry to say that, but... CommunityBridge is some kind of complicated sh*t, to support project with donation... You can't just simply "donate". You need CB account, then you will have to add your credit card and then you might donate... Only Credit Cards are supported, no PayPal or other kind of digital payment platforms (apple pay, google pay, european bank trasfer). Are you really want to raise some funds by making it all that hard?

**Comment by bitalianz** [comment]
Hello,

Im new with RethinkDB, but it's been an amazing experience working with it. I wonder if it could be possible to anounce the linux foundation link (https://funding.communitybridge.org/projects/5a679b67-5b2f-4d4d-9c4d-6fc165daaa76) in RethinkDB landing page?

Thanks for this great effort.

**Comment by gabor-boros** [comment]
Dear @bitalianz,
Thank you for your question! With @KittyBot we are working on a blog post/announcement which will contain this information too.

**Comment by cgstudio-in** [comment]
@gabor-boros @KittyBot @srh Thanks a lot for bringing back my confidence guys. I'm working on a electron based app with rethinkdb. I'll write a boilerplate electron, react, graphQL, rethinDB boilerplate so that the most popular question "Best db for electron apps" will get a answer with a kick start solution.

I'll be waiting to hear from you. Good luck...

**Comment by ChrisTalman** [comment]
Will ReQL-on-FoundationDB be discussed in the blog post? I'm still hopeful that can get off the ground at some point.

**Comment by srh** [comment]
No, it won't (I presume?).  I'm working down the last 2.4 milestone issues.  The only real one left is this: #6437

**Comment by Thiploll** [comment]
Why is https://funding.communitybridge.org/projects/rethinkdb not loading for me

**Comment by gabor-boros** [comment]
@Thiploll, I have an account there too. It happens to me when I automatically get logged out from the site, and I need to log back. After I logged back It loads for me.

**Comment by ChrisTalman** [comment]
I didn't realise until now, but, for anyone like me who may have not yet seen it, it seems that the official blog post from @gabor-boros and @KittyBot [has been posted](https://rethinkdb.com/blog/community-update-stayin-alive).

**Comment by gabor-boros** [comment]
Hey everyone :wave: As promised, we've got [technical updates](https://rethinkdb.com/blog/technical-update-keep-the-ball-rolling) for you! 
We've got a few more updates to share later today and will be around to answer any questions here or on Twitter, HN, Slack, Discord, etc.

**Comment by tinco** [comment]
Hi @gabor-boros! What's the best place to keep track of what's going on development/milestone wise as a maintainer? I'd like to make sure the Ruby is up to speed with anything you guys might be working on,  I'm not sure anyone even remembers I am a (the?) ruby driver maintainer. I don't know if anyone knows how to reach me if they want something fixed in the Ruby driver as I don't actively browse the issues. Maybe I can subscribe to notifications on issues with a certain label or something?

**Comment by gabor-boros** [comment]
Hello @tinco 👋 
Yes I remember 😊 

Right now I put together a GitHub [project ](https://github.com/orgs/rethinkdb/projects/1) but is for tracking leftovers around separate repos.

Correct me if I’m wrong but as I remember the ruby driver is compatible with the 2.4.x release am I right?

> I don't know if anyone knows how to reach me if they want something fixed in the Ruby driver

I think GitHub issues would be good or at least the community members used to open issues as I can see (in other repos). Also it is a question to me how can I reach you about not fix related things 😅 are you on Slack?

**Comment by gabor-boros** [comment]
For those who did not seen it:

> Sorry to interrupt :slightly_smiling_face: I just wanted to share that we've opened up a PR to discuss the governance model for RethinkDB and we would love everyone's input. I'll pin this to the top of the channel, too. Here's the tweet, too, if you'd like to share. :slightly_smiling_face: Thanks!

https://twitter.com/rethinkdb/status/1197584153681825793

Cheers 🥂:tada:

**Comment by Bessonov** [comment]
Worth mention the effort to run rethinkdb on kubernetes:
https://github.com/jmckind/rethinkdb-operator

**Comment by gabor-boros** [comment]
Exactly! We are trying to collect everything before the next release, and this is on my list as well, to (at least) mention the project on the website. Personally, I would like to have bigger support for cloud-native tech stacks.

**Comment by srh** [comment]
What do you mean you're trying to collect everything?  What needs to be done before the next release?

**Comment by gabor-boros** [comment]
In my opinion the following is needed (not complete list since I’m tired):
- Release notes
- Fixing everything in the 2.4.x GitHub project
- Fixing homebrew releasing
- Contact driver maintainers
- Release new driver versions (or be able to release them)
- Update the docs (with your PR)
- Update examples
- Add new 2.4.x specific examples
- Update Dockerfiles
- Update DigitalOcean image

Generally these are in my head, but I’ll organize these and solve them until the end of the week

**Comment by wtfp2p** [comment]
is the donation page still working?

**Comment by gabor-boros** [comment]
@wtfp2p What do you mean? This is the donation page we have 😊 https://funding.communitybridge.org/projects/rethinkdb

**Comment by wanjochan** [comment]
@gabor-boros  the website (https://funding.communitybridge.org/projects/rethinkdb) return "Failed to create card." when add credit card, and no response from their help-desk...

**Comment by gabor-boros** [comment]
@wanjochan Ohh that is really unfortunate 😟 I’ll contact LF about what should be done in cases like yours. I’ll keep you informed about this

**Comment by agnies-w** [comment]
Hello @wanjochan and @gabor-boros - We've been receiving regular donations to projects on CommunityBridge. I just spoke with our Helpdesk and we don't have a support ticket from you. Please file a support ticket with our help desk at https://jira.linuxfoundation.org/servicedesk/customer/portal/4 and provide more details on the issue. It's best to interact with our Helpdesk via JIRA to not expose personal info on GitHub issues. Thank you.

**Comment by ShubhraKar** [comment]
Hey folks, I just made a donation using credit card. I logged in with my GitHub ID and added my card. Worked for me fine. It did prompt me to enter my zip and CVC code. Maybe because I am in a US zip code, there isn't an issue ?

**Comment by AaronRobinson** [comment]
Could everyone else try a donation too? Hopefully they’ll all work :)

**Comment by wanjochan** [comment]
@agnies-w  new ticket created (  https://jira.linuxfoundation.org/servicedesk/customer/portal/2/IT-18580 ) and I've just tried again on ( https://funding.communitybridge.org/projects/rethinkdb ) it remains failed, which my cards (tried Visa or Mastercard) are sure working fine.

<img width="1226" alt="WeChate4f9f663dc6d5271b9b798eb7f14a266" src="https://user-images.githubusercontent.com/6470068/71217854-2835bc00-22fa-11ea-9ad8-d97ce8339e07.png">

**Comment by wanjochan** [comment]
@ShubhraKar  I've tried my Visa issued from China and Mastercard issued from Australia, which cards I used all the time without any problem.  From the errmsg "Failed to create card." I really got no ideas...

**Comment by ShubhraKar** [comment]
@wanjochan the LF support team is looking into it. Thanks for creating the ticket. Definitely some bug or a stripe limitations as 3 donations were made today successfully. We will chase it down. Hang on.

**Comment by gabor-boros** [comment]
Since this is under investigation by LF (as @ShubhraKar mentioned), and the **new release is out**, I'm closing this issue. For those who missed it and wondering what I'm talking about: https://twitter.com/rethinkdb/status/1207730230175191043?s=20

Based on Stuart's suggestion in the twitter comments see my next comment.

**Comment by gabor-boros** [comment]
**No.**

**Comment by triethyl** [comment]
I have just started looking for an open-source real-time database and rethinkdb is what the entire internet seems to be talking about. But I am also reading the project is supposed to be dead yet there are some commits over the last 2 months.

I hope this project doesn't go dead. It's an amazing one.

**Comment by Thiploll** [comment]
> I have just started looking for an open-source real-time database and rethinkdb is what the entire internet seems to be talking about. But I am also reading the project is supposed to be dead yet there are some commits over the last 2 months.
> 
> I hope this project doesn't go dead. It's an amazing one.

The project was re-opened not so long ago, the makers are looking for some funding.

**Comment by gabor-boros** [comment]
@Thiploll and @triethyl Thank you for caring about RethinkDB ❤️ 

I would like to point out that everything slowed down due to the situation around the world, but we still care about the project too and waiting for contributors to help us. 😊

**Comment by v3ss0n** [comment]
appwrite , supabase  - none of them compare to RethinkDB . This project only stalled because the company died . Their engineers are great people.
@srh  , @gabor-boros  and a few other community is trying to revive it. Please support all you can.

**Comment by gabor-boros** [comment]
I think the best support right now is helping to maintain the drivers and the DB 😇

**Comment by varaskkar** [comment]
I hope it will be reborn...

**Comment by devinrhode2** [comment]
I've always been pretty interested in Rethinkdb for it's ability to create uniquely great user experiences

Also over the past few years, I've really learned typescript and gained an appreciation for the benefits of static typing

RethinkDB is a NoSQL, schema-less json data store.

As static typing continues to gain popularity, I predict a schema-less data storage solution will proportionally shrink

**Comment by withinboredom** [comment]
> As static typing continues to gain popularity, I predict a schema-less data storage solution will proportionally shrink

It comes and goes. static typing / non-static typing becomes popular/unpopular over the decades. They have their uses and benefits one way or the other, and sometimes types are benefits; sometimes they work against you.

**Comment by AaronRobinson** [comment]
Document databases came about when statically typed languages were
ubiquitous so I don’t think the introduction of types in one language is
going to make a difference.

**Comment by withinboredom** [comment]
Yes, I'd also add that the benefit to nosql is that you don't have to spend too much time/cycles (de)serializing things like you do with regular SQL. A large part of an ORM is specifically dedicated to that (as well as turning your objects in a schema). With nosql, you just serialize state and move most of that logic into your application. In some ways, that can be better.

Similar to how if you are doing a lot of side-effects outside your system, you might switch off transactions so that an unrelated error won't rollback your transaction after you've performed side-effects. It's all tradeoffs.
